File: java_app/src/main/java/me/halin/android_architecture/main/MainActivity.java
package me.halin.android_architecture.main;

import android.arch.lifecycle.Observer;
import android.arch.lifecycle.ViewModelProviders;
import android.content.Intent;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;

import me.halin.android_architecture.R;
import me.halin.android_architecture.databinding.ActivityMainBinding;
import me.halin.android_architecture.main.model.MainViewModel;
import me.halin.android_architecture.other.OtherActivity;

public class MainActivity extends AppCompatActivity {

    private MainViewModel viewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        this.viewModel = ViewModelProviders.of(this).get(MainViewModel.class);

        ActivityMainBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        binding.setLifecycleOwner(this);
        binding.setViewModel(this.viewModel);

        // 构造Adapter并初始化binding
        MainAdapter adapter = new MainAdapter(this.getLayoutInflater());
        binding.recyclerView.setLayoutManager(new LinearLayoutManager(this));
        binding.recyclerView.setAdapter(adapter);

        // 实现监听
        this.viewModel.versionModel.observe(this, adapter);

        this.viewModel.openOtherEvent.observe(this, new Observer<Void>() {
            @Override
            public void onChanged(@Nullable Void aVoid) {
                Intent intent = new Intent(MainActivity.this, OtherActivity.class);
                intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
                MainActivity.this.startActivity(intent);
            }
        });
    }

    @Override
    protected void onResume() {
        super.onResume();
    }
}

File: app/src/main/java/me/halin/android_architecture/main/MainActivity.kt
package me.halin.android_architecture.main

import android.arch.lifecycle.Observer
import android.arch.lifecycle.ViewModelProviders
import android.content.Intent
import android.databinding.DataBindingUtil
import android.net.Uri
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import me.halin.android_architecture.R
import me.halin.android_architecture.databinding.ActivityMainBinding
import me.halin.android_architecture.main.model.MainViewModel

class MainActivity : AppCompatActivity() {

    lateinit var viewModel: MainViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        this.viewModel = ViewModelProviders.of(this).get(MainViewModel::class.java)

        val binding = DataBindingUtil.setContentView<ActivityMainBinding>(this, R.layout.activity_main)
        binding.setLifecycleOwner(this)
        binding.viewModel = this.viewModel

        // 构造Adapter并初始化binding
        val adapter = MainAdapter(this.layoutInflater)
        binding.recyclerView.layoutManager = LinearLayoutManager(this)
        binding.recyclerView.adapter = adapter

        viewModel.versionModel.observe(this, adapter)
        this.viewModel.openOtherEvent.observe(this, Observer { _ ->
            val intent = Intent(Intent.ACTION_VIEW)
            intent.data = Uri.parse("http://www.baidu.com")
            this.startActivity(intent)
        })
    }
}
